lib/react.dart

```diff
@@ -748,8 +754,27 @@ abstract class Component2 implements Component {
   /// Invoked once before the `Component` is mounted. The return value will be used as the initial value of [state].
   ///
   /// See: <https://facebook.github.io/react/docs/react-component.html#getinitialstate>
+  ///
+  /// > __DEPRECATED - DO NOT USE__
+  /// >
+  /// > Use [init] or [getDerivedStateFromProps] to handle intialization instead
+  @Deprecated('6.0.0')
   Map getInitialState() => const {};
 
+  /// This is equivalent to `Constructor` in React 16, this is called before mounting
+  /// See: <https://reactjs.org/docs/react-component.html#constructor>
+  ///
+  /// __Example__:
+  ///
+  ///    class MyClass extends react.Component2 {
+  ///      Map initialState = { foo: 0, bar: 1};
```

Comment:
There's no real value in suggesting that consumers store this as a separate variable.  Should just assign `this.state` to the map within `init` like so:

```dart
@override
void init() {
  this.state = {'foo': 0, 'bar': 1};
}
```